api/v1alpha1/api.go

```diff
@@ -435,6 +435,50 @@ type BackendSecurityPolicyAWSCredentials struct {
 	//
 	// +optional
 	OIDCExchangeToken *AWSOIDCExchangeToken `json:"oidcExchangeToken,omitempty"`
+
+	// Rotation specifies the configuration for rotating AWS credentials.
+	// Only applies when CredentialsFile is used.
+	// +optional
+	Rotation *AWSCredentialsRotation `json:"rotation,omitempty"`
+}
+
+// AWSCredentialsRotation specifies whether and how to rotate AWS credentials
+type AWSCredentialsRotation struct {
```

Comment:
I feel we do not necessarily need to expose this API to user, this is controller implementation details for rotating credentials and it is also not specific to AWS.